apollo-compiler/src/test/graphql/com/example/fragment_friends_connection/FragmentFriendsConnectionExpected.java

```diff
@@ -1,15 +1,56 @@
 package com.example.fragment_friends_connection;
 
+import com.apollostack.api.GraphQLQuery;
+import java.lang.Override;
+import java.lang.String;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import javax.annotation.Nullable;
 
-public interface FragmentFriendsConnection {
-  @Nullable Hero hero();
+public final class FragmentFriendsConnection implements GraphQLQuery<FragmentFriendsConnection.Data> {
```

Comment:
First: want to override `com.apollostack.api.GraphQLQuery#getOperationDefinition` and `com.apollostack.api.GraphQLQuery#getFragmentDefinitions`, as well as in future it will hold argumetns. So `Apollo` client will deal only with `GraphQLQuery` to get everything required to run query.
Second there is no need for user to create impl of this query, only for response.